#!/usr/bin/env python
PACKAGE = "robocup_object_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("diameter_soccer_ball", double_t, 0, "diameter of soccer ball", 0.15, 0, 1)
gen.add("ball_detection_range", double_t, 0, "range of ball detection(m)", 3, 0, 5)
gen.add("width_robot", double_t, 0, "width of robot", 0.3, 0, 1)
gen.add("height_robot", double_t, 0, "height of robot", 0.6, 0, 1)
gen.add("max_robot_detection_range", double_t, 0, "maximum range of robot detection(m)", 3, 0, 5)
gen.add("min_robot_detection_range", double_t, 0, "minimum range of robot detection(m)", 0.3, 0, 5)
gen.add("bibs_height", double_t, 0, "height of bibs size", 0.15, 0, 1)
gen.add("bibs_min_aspect_ratio", double_t, 0, "minmum aspect ratio(height/width) of bibs", 0.8, 0, 5)

gen.add('view_angle_width', double_t, 0, "Width of view angle(deg)", 40, 0, 360)
gen.add('view_angle_height', double_t, 0, "height of view angle(deg)", 80, 0, 360)


exit(gen.generate(PACKAGE, "robocup_object_detector", "Params_object_detector"))
